singer/transform.py

```diff
@@ -262,6 +263,8 @@ def _transform(self, data, typ, schema, path):
         if typ == "null":
             if data is None or data == "":
                 return True, None
+            elif isinstance(data, decimal.Decimal) and data.is_nan():
```

Comment:
This should include `is_snan` and `is_qnan`.

Reply:
So according to [these](https://docs.python.org/3/library/decimal.html#decimal.Decimal.is_nan)  docs, it seems like the `is_nan` will catch everything that `is_snan` and `is_qnan` captures.
Is there any reason we're not thinking of that we'd wanna still include the `is_snan` or `is_qnan`?

Reply:
Oh, good. That makes sense. Looks like internally, decimal uses only `is_nan` when it needs to check for math operations. E.g., https://github.com/python/cpython/blob/4e0ce820586e93cfcefb16c2a3df8eaefc54cbca/Lib/_pydecimal.py#L1886

Reply:
Given the comment below, though, it's probably best to just remove this and not bring this NaN idea into the `"null"` schema handling unless we have a real reason to do so, since it is technically losing information to go from `NaN/+Infinity/-Infinity/None` all to just `None`.